### PR TITLE
[Tts][TCSACR-497] Add new API for service state information

### PR DESCRIPTION
**Pull request by stom-hwang**
Signed-off-by: Suyeon Hwang <stom.hwang@samsung.com>

### Description of Change ###

Add new API for service state information

### API Changes ###

 - ACR: https://code.sec.samsung.net/jira/browse/TCSACR-497

### CODE REVIEW

File: src/Tizen.Uix.Tts/Tizen.Uix.Tts/TtsClient.cs [hunk 1]
```diff
@@ -203,6 +203,37 @@ public enum Voice
...
+    /// <summary>
+    /// Enumeration for the states of TTS service.
+    /// </summary>
+    /// <since_tizen> 10 </since_tizen>
+    public enum ServiceState
+    {
+        /// <summary>
+        /// Ready state.
+        /// </summary>
+        /// <since_tizen> 10 </since_tizen>
```
Comment (wiertel) [inline src/Tizen.Uix.Tts/Tizen.Uix.Tts/TtsClient.cs:215]: you don't need the \<since_tizen\> in enum field if the field was introduced in the initial version of the enum - i.e. if the version here is the same as the \<since_tizen\> of the containing enum

File: src/Tizen.Uix.Tts/Tizen.Uix.Tts/TtsClient.cs [hunk 2]
```diff
@@ -589,6 +623,51 @@ public TtsClient()
             }
         }
 
+        /// <summary>
+        /// Event to be invoked when state of TTS service changes.
```
Comment (wiertel) [inline src/Tizen.Uix.Tts/Tizen.Uix.Tts/TtsClient.cs:627]: just a suggestion:
```suggestion
        /// Event to be invoked when the state of TTS service changes.
```

File: src/Tizen.Uix.Tts/Tizen.Uix.Tts/TtsClient.cs [hunk 3]
```diff
@@ -673,6 +752,32 @@ public State CurrentState
...
+        /// <summary>
+        /// Gets the current state of TTS service.
+        /// </summary>
+        /// <since_tizen> 10 </since_tizen>
+        /// <value>
+        /// The current state of TTS service.
+        /// </value>
+        /// <returns>
+        /// Current TTS service state value.
+        /// </returns>
```
Comment (wiertel) [inline src/Tizen.Uix.Tts/Tizen.Uix.Tts/TtsClient.cs:764]: \<returns\> for a property is ignored as not related to this element, you can remove those 3 lines

File: src/Tizen.Uix.Tts/Tizen.Uix.Tts/TtsClient.cs [hunk 4]
```diff
@@ -203,6 +203,37 @@ public enum Voice
...
+        /// Synthesizing state.
+        /// </summary>
+        /// <since_tizen> 10 </since_tizen>
+        Synthesizing = 1,
+
+        /// <summary>
+        /// Playing state.
+        /// </summary>
+        /// <since_tizen> 10 </since_tizen>
+        Playing = 2,
```
Comment (wiertel) [inline src/Tizen.Uix.Tts/Tizen.Uix.Tts/TtsClient.cs:228]: just wondering - can those state overlap? i.e. parallel synth and stream play in real time?

File: src/Tizen.Uix.Tts/Tizen.Uix.Tts/TtsClient.cs (hunk 1 above)
Comment (stom-hwang) [reply src/Tizen.Uix.Tts/Tizen.Uix.Tts/TtsClient.cs]: Thank you for checking. I will remove it soon.

File: src/Tizen.Uix.Tts/Tizen.Uix.Tts/TtsClient.cs (hunk 4 above)
Comment (stom-hwang) [reply src/Tizen.Uix.Tts/Tizen.Uix.Tts/TtsClient.cs:225]: In the aspect of real behavior of TTS service engine, yes it is possible to parallely synth and stream play in some times.
We just want to seperate the timing between the begining of the synthesis and the begining of the playing. That is, `Synthesizing` state means that the TTS service engine only synthesizes the sound data without playing.
TTS service engine will play sound data when the synthesized data is enough to play and then the service state will be changed to `Playing`.

File: src/Tizen.Uix.Tts/Tizen.Uix.Tts/TtsClient.cs (hunk 3 above)
Comment (stom-hwang) [reply src/Tizen.Uix.Tts/Tizen.Uix.Tts/TtsClient.cs]: Thank you for checking. I will remove it soon.

File: src/Tizen.Uix.Tts/Tizen.Uix.Tts/TtsClient.cs (hunk 2 above)
Comment (stom-hwang) [reply src/Tizen.Uix.Tts/Tizen.Uix.Tts/TtsClient.cs]: Thank you for suggestion. I will fix it soon.

**Review by stom-hwang — COMMENTED** [review]
Thank you for checking.
I will fix it.

**Review by syeonholic — APPROVED** [review]
It looks good to me.

**Review by syeonholic — APPROVED** [review]
It looks good to me.

**Review by wonnamy — APPROVED** [review]
It looks good to me.

**Review by urmain — APPROVED** [review]
LGTM

**Pull request merged**